graphrag/query/structured_search/global_search/search.py

```diff
@@ -176,7 +176,7 @@ async def _map_response_single_batch(
             search_prompt = self.map_system_prompt.format(context_data=context_data)
             search_messages = [
                 {"role": "system", "content": search_prompt},
-                {"role": "user", "content": query},
+                {"role": "user", "content": query + "\n```json\n"}
```

Comment:
To enable merging, I'll remove this line and test it for a following PR. Can you please describe why is this needed?

Reply:
Global search supports (and prefers actually) the JSON mode, so I wonder if this may break it in JSON mode

Reply:
I have observed multiple definitions of BaseLLM within the project. My understanding is that graphrag.llm.base.base_llm.BaseLLM and graphrag.query.llm.base.BaseLLM are used in different conversational contexts. The former (graphrag.llm.base.base_llm.BaseLLM) is used for text generation, while the latter (graphrag.query.llm.base.BaseLLM) is designed for multi-turn chat.

The identifiers in the JSON output are special tokens intended for compatibility with other large models, and you can safely remove them if needed.